spec/amazon_ses_mailer/base_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+RSpec.describe AmazonSesMailer::Base do
+    let(:template_name) { 'welcome_email' }
+    let(:options) { { template: template_name } }
+    let(:delivery_method) { nil }
+    subject { described_class.new(template_name) }
+  
+    describe '.mail' do
+      before do
+        allow(AmazonSesMailer::Base).to receive(:delivery_method)
+          .and_return(delivery_method)
+      end
+  
+      it 'creates a new message' do
+        expect(AmazonSesMailer::Message).to receive(:new)
+        subject.mail(options)
+      end
+    end
+  
+    describe '.deliveries' do
+      context 'when there are no deliveries' do
+        it 'should return empty array' do
+          expect(AmazonSesMailer::Base.deliveries).to eq([])
+        end
+      end
+  
+      context 'when there are deliveries' do
+        let(:delivery_method) { :test }
+        before do
+          allow(AmazonSesMailer::Base).to receive(:delivery_method)
+            .and_return(delivery_method)
+        end
+        it {
+          expect { AmazonSesMailer::Base.mail(options).deliver }
+            .to change { AmazonSesMailer::Base.deliveries.count }.by(1)
+        }
+      end
+    end
+    describe '.transform_array' do
+      context 'when given object is array' do
+        it 'should convert false/nil into empty string and all others into string' do
```

Comment:
This is not a unit test. It should test that it calls `transform_value` method on each array member (which you should stub).